app/repository/RiderRankingRepositoryImpl.java

```diff
@@ -29,66 +26,90 @@ public RiderRankingRepositoryImpl(JPAApi jpaApi, DatabaseExecutionContext databa
     }
 
     @Override
-    public CompletionStage<Stream<RiderRanking>> getAllRiderRankings() {
-        return supplyAsync(() -> wrap (this::getAllRiderRankings), databaseExecutionContext);
+    public CompletionStage<Stream<RiderRanking>> getAllRiderRankings(long riderStageConnectionId) {
+        return supplyAsync(() -> wrap (entityManager -> getAllRiderRankings(entityManager, riderStageConnectionId)), databaseExecutionContext);
     }
 
-    private Stream<RiderRanking> getAllRiderRankings(EntityManager em){
-        List<RiderRanking> riderRankings = em.createQuery("select rR from RiderRanking rR", RiderRanking.class).getResultList();
-        return riderRankings.stream();
+    private Stream<RiderRanking> getAllRiderRankings(EntityManager em, long riderStageConnectionId){
+        TypedQuery<RiderRanking> query = em.createQuery("select rR from RiderRanking rR where rR.riderStageConnection.id = :riderStageConnectionId" , RiderRanking.class);
+        query.setParameter("riderStageConnectionId", riderStageConnectionId);
+        return query.getResultList().stream();
     }
 
     @Override
-    public CompletionStage<Stream<RiderRanking>> getAllRiderRankingsByType(RankingType rankingType) {
-        return supplyAsync(() -> wrap (em -> getAllRiderRankingsByRankingType(em, rankingType)), databaseExecutionContext);
+    public CompletionStage<Stream<RiderRanking>> getAllRiderRankingsByType(long riderStageConnectionId, String rankingType) {
+        return supplyAsync(() -> wrap (em -> getAllRiderRankingsByRankingType(em, riderStageConnectionId, rankingType)), databaseExecutionContext);
     }
 
-    private Stream<RiderRanking> getAllRiderRankingsByRankingType(EntityManager em, RankingType rankingType){
-        TypedQuery<RiderRanking> query = em.createQuery("select rR from RiderRanking rR where rR.rankingType = :rankingType" , RiderRanking.class);
-        query.setParameter("rankingType", rankingType);
+    private Stream<RiderRanking> getAllRiderRankingsByRankingType(EntityManager em, long riderStageConnectionId, String rankingType){
+        TypedQuery<RiderRanking> query = em.createQuery("select rR from RiderRanking rR where rR.riderStageConnection.id = :riderStageConnectionId and rR.rankingType = :rankingType" , RiderRanking.class);
+        query.setParameter("riderStageConnectionId", riderStageConnectionId);
+        query.setParameter("rankingType", RankingType.valueOf(rankingType));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "rankingType" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)